### DISCUSSION TITLE
Feature Request: Support for beta mermaid diagrams

### DISCUSSION THREAD

**Discussion by brunopacheco1** [thegeeklab/hugo-geekdoc#797] [Ideas]
## Summary

Add support for beta mermaid diagrams.

## Problem

Current version of the theme does not support block-beta, for example:
<img width="1002" alt="image" src="https://github.com/thegeeklab/hugo-geekdoc/assets/4491850/42ff23d6-39fa-4c46-9ff2-25a93a09b16e">

it would be interesting to be able to use experimental features.

I want to contribute, btw, but I am not sure if this was a deliberated decision. Please let me, so I can investigate on how to do it.

**Comment by xoxys** [comment]
Thanks for your report. In general, I'm not against it, but it depends a bit on what's required. From searching the mermaid docs, I have not found how experimental features can be enabled. Do you have a ref?

**Reply by brunopacheco1** [reply]
@xoxys thanks for taking care of this. I don't know either, perhaps it's related to version of mermaid.js that geekdoc uses.

I tried to use https://mermaid.js.org/syntax/block.html in hugo-geekdoc and the output is the one I shared in the screenshot, whereas other places, like GitHub, they are presented correctly. Please check the links below.

Experimental/beta Mermaid.js diagrams deployed with hugo geekdoc: https://genomicdatainfrastructure.github.io/gdi-userportal-docs/docs/test/

Experimental/beta Mermaid.js diagrams visible in GitHub: https://github.com/GenomicDataInfrastructure/gdi-userportal-docs/blob/main/content/docs/test/_index.md

**Reply by xoxys** [reply]
Thanks. Block Diagrams were added in mermaid v10.8 while the latest release of hugo-geedoc is using v10.7. I've tagged a new release of the theme, which includes mermaid v 10.9 now.
